test/test_torch.py

```diff
@@ -12541,13 +12541,40 @@ def test_has_internal_overlap(self):
         self.assertEqual(torch._debug_has_internal_overlap(b_expanded), OVERLAP_YES)
 
     @staticmethod
-    def unary_check_mem_overlap(self, inplace_op, value=-0.5, device='cpu'):
+    def unary_check_internal_mem_overlap(self, inplace_op, value=-0.5,
+                                         device='cpu'):
         tensor = torch.tensor(value, device=device).expand(3, 3)
         with self.assertRaisesRegex(RuntimeError, 'single memory location'):
             inplace_op(tensor)
 
     @staticmethod
-    def binary_check_mem_overlap(self, inplace_op, value=-0.5, device='cpu'):
+    def unary_check_input_output_mem_overlap(self, data, sz, op):
+
+        def _test(op, c, a):
```

Comment:
Two notes here:

First, because the dispatch for op(a) and op(a, out=c_exp) might be different, we should test this differently.
```
def _test(op, output, input):
  output_exp = torch.empty_like(output)
  op(output, out=c_exp) 
  self.assertEqual(op(input, out=output), output_exp, op.__name__)
```

Secondly, each of the below cases can be written as (for example)
```
# c is identical to a:
_test(op, output=data[0:sz], output=data[0:sz])
```
which is much more compact.

Reply:
addressed